pkg/kn/commands/service_describe.go

```diff
@@ -16,53 +16,120 @@ package commands
 
 import (
 	"errors"
+	"fmt"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"io"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"text/tabwriter"
+	"time"
 
 	"github.com/spf13/cobra"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
 func NewServiceDescribeCommand(p *KnParams) *cobra.Command {
-
-	serviceDescribePrintFlags := genericclioptions.NewPrintFlags("").WithDefaultOutput("yaml")
-	serviceDescribeCommand := &cobra.Command{
+	command := &cobra.Command{
 		Use:   "describe NAME",
-		Short: "Describe available services.",
+		Short: "Show details for a given service",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) < 1 {
-				return errors.New("requires the service name.")
+				return errors.New("no service name provided")
 			}
-			client, err := p.ServingFactory()
-			if err != nil {
-				return err
+			if len(args) > 1 {
+				return errors.New("more than one service name provided")
 			}
-
 			namespace, err := GetNamespace(cmd)
 			if err != nil {
 				return err
 			}
-			describeService, err := client.Services(namespace).Get(args[0], v1.GetOptions{})
-			if err != nil {
-				return err
-			}
-
-			printer, err := serviceDescribePrintFlags.ToPrinter()
-			if err != nil {
-				return err
-			}
-			describeService.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
-				Group:   "knative.dev",
-				Version: "v1alpha1",
-				Kind:    "Service"})
-			err = printer.PrintObj(describeService, cmd.OutOrStdout())
+			service, err := getService(namespace, args[0], p)
 			if err != nil {
 				return err
 			}
+			describeService(cmd.OutOrStdout(), service)
 			return nil
 		},
 	}
-	AddNamespaceFlags(serviceDescribeCommand.Flags(), false)
-	serviceDescribePrintFlags.AddFlags(serviceDescribeCommand)
-	return serviceDescribeCommand
+	AddNamespaceFlags(command.Flags(), false)
+	return command
+}
+
+func describeService(w io.Writer, service *v1alpha1.Service) {
+	dw := newDescribeWriter(w)
+	defer dw.flush()
+
+	dw.printlnWithLabel("Name", service.Name)
+	dw.printlnWithLabel("Namespace", service.Namespace)
+	dw.printlnWithLabel("Address", service.Status.Address.Hostname)
+	dw.printlnWithLabel("Domain", service.Status.Domain)
+	dw.printlnWithLabel("Age", dw.age(service.CreationTimestamp.Time))
+	dw.newline()
+
+	/*
+	   Additional information which might make sense to add:
+	   - Revisions overview (like image)
+	   - Traffic distribution
+	   - Timeout seconds
+	   - Conditions
+	   - Relevant Events (if any)
+	   - Annotations / Labels
+	   - Reference to configuration
+	*/
+}
+
+func getService(namespace string, name string, p *KnParams) (*v1alpha1.Service, error) {
```

Comment:
We'll be using GetService in multiple commands, this is also a target for including in some package. Fine keeping it here for now.